packages/ord-connect/src/hooks/useSend.tsx

```diff
@@ -1,61 +1,73 @@
-import { useState } from "react";
+import { useCallback, useState } from "react";
 import { JsonRpcDatasource, PSBTBuilder } from "@ordzaar/ordit-sdk";
 
 import signPsbt from "../lib/signPsbt";
-import { useOrdContext } from "../providers/OrdContext";
+import { useOrdConnect } from "../providers/OrdConnectProvider";
 
 type SendFunction = (
   address: string,
   satoshis: number,
-  feeRate?: number,
+  feeRate: number,
 ) => Promise<string | null>;
 
-export function useSend(): [SendFunction, string | null, boolean] {
-  const { wallet, network, address, publicKey } = useOrdContext();
+export function useSend() {
+  const { wallet, network, address, publicKey } = useOrdConnect();
   const [error, setError] = useState<string | null>(null);
   const [loading, setLoading] = useState<boolean>(false);
 
-  const datasource = new JsonRpcDatasource({ network });
+  const send: SendFunction = useCallback(
+    async (toAddress, satoshis, feeRate) => {
+      setLoading(true);
+      try {
+        setError(null);
+        if (
+          !address ||
+          !address.payments ||
+          !publicKey ||
+          !publicKey.payments ||
+          !wallet
+        ) {
+          throw new Error("No wallet is connected");
+        }
 
-  const send: SendFunction = async (toAddress, satoshis, feeRate) => {
-    setLoading(true);
-    try {
-      setError(null);
-      if (!address || !publicKey) {
-        throw new Error("No wallet is connected");
+        const psbtBuilder = new PSBTBuilder({
+          address: address.payments,
+          feeRate,
+          network,
+          publicKey: publicKey.payments,
+          outputs: [
+            {
+              address: toAddress,
+              value: satoshis,
+            },
+          ],
+        });
+        await psbtBuilder.prepare();
+
+        const signedPsbt = await signPsbt({
+          address: address.payments,
+          wallet,
+          network,
+          psbt: psbtBuilder.toPSBT(),
+        });
+
+        if (!signedPsbt.hex) {
+          throw new Error("PSBT is not signed correctly");
```

Comment:
let's use a consistent error message, saw that `useSign` has this check too, or should we handle at `signPsbt`?

Reply:
Removed this since `hex` won't be null, which passes type checks. Also let ordit-sdk throw the error instead since it already checks if it's empty there